controllers/antreainstall_controller.go

```diff
@@ -216,6 +142,122 @@ func (r *AntreaInstallReconciler) Reconcile(request ctrl.Request) (ctrl.Result,
 			}
 		}
 	}
+	return reconcile.Result{}, nil
+}
+
+func (k8s *AdaptorK8s) Reconcile(r *AntreaInstallReconciler, request ctrl.Request) (ctrl.Result, error) {
+	reqLogger := r.Log.WithValues("Request.NamespacedName", request.NamespacedName)
+	if request.Namespace == "" && request.Name == operatortypes.ClusterConfigName {
+		reqLogger.Info("Reconciling antrea-operator Cluster Network CR change")
+	} else if request.Namespace == operatortypes.OperatorNameSpace && request.Name == operatortypes.OperatorConfigName {
+		reqLogger.Info("Reconciling antrea-operator antrea-install CR change")
+	} else {
+		return reconcile.Result{}, nil
+	}
+
+	// Fetch antrea-install CR.
+	operConfig := &operatorv1.AntreaInstall{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: operatortypes.OperatorNameSpace, Name: operatortypes.OperatorConfigName}, operConfig)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			msg := fmt.Sprintf("%s CR not found", operatortypes.OperatorConfigName)
+			log.Info(msg)
+			r.Status.SetDegraded(statusmanager.ClusterConfig, "NoAntreaInstallCR", msg)
+			return reconcile.Result{}, nil
+		}
+		log.Error(err, "failed to get antrea-install CR")
+		r.Status.SetDegraded(statusmanager.OperatorConfig, "InvalidAntreaInstallCR", fmt.Sprintf("Failed to get operator CR: %v", err))
+		return reconcile.Result{Requeue: true}, err
+	}
+	if request.Name == operConfig.Name && r.AppliedOperConfig != nil {
+		if reflect.DeepEqual(operConfig.Spec, r.AppliedOperConfig.Spec) {
+			log.Info("no configuration change")
+			return reconcile.Result{}, nil
+		}
+	}
+
+	// Apply configuration.
+	if result, err := applyConfig(r, k8s.Config, nil, operConfig, nil); err != nil {
+		return result, err
+	}
+
+	r.Status.SetNotDegraded(statusmanager.ClusterConfig)
+	r.Status.SetNotDegraded(statusmanager.OperatorConfig)
+
+	r.AppliedOperConfig = operConfig
+
+	return ctrl.Result{}, nil
```

Comment:
Why here returned `ctrl.Result{}` while at line 175 it returned `reconcile.Result{}`? Are they similar or different?

Reply:
Nice reminder, they are actually the same as `ctrl.Result{}` is declared as `type Result = reconcile.Result`. There should be a mistype, to be consistent, `ctrl.Result{}` will be removed, and only `reconcile.Result{}` will be used.

Reply:
> Why here returned `ctrl.Result{}` while at line 175 it returned `reconcile.Result{}`? Are they similar or different?

Nice reminder, they are actually the same as ctrl.Result{} is declared as type Result = reconcile.Result. There should be a mistype, to be consistent, ctrl.Result{} will be removed, and only reconcile.Result{} will be used.